### PR TITLE
Switch to GitHub Actions

### PR DESCRIPTION
**Pull request by Fryguy**
Travis is no longer working, but GitHub Actions does!

This PR:
- Switches to GitHub Actions.
- Adds newer Rubies and Sidekiq versions to the test matrix
- Drops code-climate-test-reporter (as that's no longer supported) and prefers paambaati/codeclimate-action

GitHub Actions will _not_ run on the main repo from the pull request that introduces it (i.e. this one), so don't expect to see it here.  Instead you can look at the branch for this PR from my fork and see the run there: 

[CI](https://github.com/Fryguy/sidecloq/actions/workflows/ci.yml) https://github.com/Fryguy/sidecloq/actions/workflows/ci.yml?query=branch%3Agithub_actions

### CODE REVIEW

**Comment by Fryguy** [comment]
One caveat - jruby builds for some reason sporadically hang...I can't tell if that's a new thing, or something it's always done.  If you "Re-run failed jobs" however, eventually they will work.

I'm wondering if it's just cleaner to mark jruby-builds to allow failures.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        - "6.4"
+        - "6.5"
+        - main
+        exclude:
+        # sidekiq 3.5 does not work with ruby >= 3
+        - ruby-version: "3.0"
+          sidekiq-version: "3.5"
+        - ruby-version: "3.1"
+          sidekiq-version: "3.5"
+        # sidekiq 6+ only supports ruby >= 2.5
```
Comment (Fryguy) [inline .github/workflows/ci.yml:38]: Dropping support for Ruby 2.4 would eliminate this section.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        - "5.2"
+        - "6.0"
+        - "6.1"
+        - "6.2"
+        - "6.3"
+        - "6.4"
+        - "6.5"
+        - main
+        exclude:
+        # sidekiq 3.5 does not work with ruby >= 3
```
Comment (Fryguy) [inline .github/workflows/ci.yml:33]: Dropping support for sidekiq 3.5 would eliminate this section.

**Comment by Fryguy** [comment]
hi @mattyr - bump?

**Review by mattyr — APPROVED** [review]
Sorry for the super long delay -- thanks for this!

**Comment by mattyr** [comment]
> One caveat - jruby builds for some reason sporadically hang...I can't tell if that's a new thing, or something it's always done. If you "Re-run failed jobs" however, eventually they will work.
> 
> I'm wondering if it's just cleaner to mark jruby-builds to allow failures.

Yeah, I recall trying to debug this obsessively for a while, and was unable to resolve... I'm kind of hoping now dropping some support for older versions could sweep that issue under the rug...

**Pull request merged**